packages/core/src/core/coreToolScheduler.ts

```diff
@@ -225,6 +227,57 @@ const createErrorResponse = (
   errorType,
 });
 
+const TRUNCATION_THRESHOLD = 1_000_000;
+const TRUNCATION_LINES = 1_000;
+
+export async function truncateAndSaveToFile(
+  content: string,
+  callId: string,
+  projectTempDir: string,
+): Promise<{ content: string; outputFile?: string }> {
+  if (content.length <= TRUNCATION_THRESHOLD) {
+    return { content };
+  }
+
+  const lines = content.split('\n');
+  let truncatedContent: string;
+
+  // Determine the truncated content for display based on its structure.
+  if (lines.length > TRUNCATION_LINES) {
+    // Content has many lines; truncate by line count.
+    truncatedContent = lines.slice(-TRUNCATION_LINES).join('\n');
+  } else {
+    // Content has few lines (or one very long line); truncate by character count.
+    const maxChars = TRUNCATION_LINES * 80; // Approximate 80 chars per line
```

Comment:
~20k tokens. I would have thought the truncation threshold would be the maxChars here though. Feels a little disconnected